src/hyperlight_host/src/crash_handler/linux.rs

```diff
@@ -0,0 +1,301 @@
+            if pattern.is_empty()
+                || pattern == "|/bin/false"
+                || pattern == "|/usr/bin/false"
+                || pattern.starts_with("|/bin/false ")
+                || pattern.starts_with("|/usr/bin/false ")
+            {
+                // Core pattern is explicitly disabled
+                return false;
+            }
+        }
+        Err(_) => {
+            // Can't read the file (permission denied, doesn't exist in container, etc.)
+            // Fail-open: assume dumps are enabled
+        }
+    }
+
+    // Check 3: Process dumpable flag
+    // This should always work - no special permissions needed
+    // Some security policies (like running setuid) disable dumping
+    unsafe {
+        let dumpable = libc::prctl(libc::PR_GET_DUMPABLE, 0, 0, 0, 0);
+        if dumpable == 0 {
+            // Process is marked non-dumpable (0 = non-dumpable)
+            return false;
+        }
+        // dumpable == 1 means normal dumpable
+        // dumpable == 2 means dumpable but with restrictions
+        // dumpable < 0 means error (very unlikely)
+        // In all these cases, continue
+    }
+
+    // All checks passed or couldn't determine - assume enabled (fail-open)
+
+    true
+}
+
+/// Initialize crash handlers for all fatal signals.
+///
+/// Called once on first sandbox registration.
+///
+/// Only installs handlers if the OS will actually generate core dumps.
+/// If core dumps are disabled, we don't install handlers since there's
+/// no point generating sandbox dumps when the host won't dump.
+pub(super) fn init_crash_handlers() {
+    // Check if core dumps are enabled
+    if !are_core_dumps_enabled() {
+        log::info!("Core dumps disabled on this system, skipping crash handler installation");
+        return;
+    }
+
+    unsafe {
+        install_handler(libc::SIGSEGV, &raw mut PREV_SIGSEGV);
+        install_handler(libc::SIGABRT, &raw mut PREV_SIGABRT);
+        install_handler(libc::SIGBUS, &raw mut PREV_SIGBUS);
+        install_handler(libc::SIGFPE, &raw mut PREV_SIGFPE);
+        install_handler(libc::SIGILL, &raw mut PREV_SIGILL);
+        install_handler(libc::SIGQUIT, &raw mut PREV_SIGQUIT);
+        install_handler(libc::SIGTRAP, &raw mut PREV_SIGTRAP);
+        install_handler(libc::SIGSYS, &raw mut PREV_SIGSYS);
+    }
+}
+
+/// Install a signal handler and save the previous handler.
+///
+/// # Safety
+///
+/// Calls unsafe libc::sigaction. Caller must ensure `prev_handler`
+/// points to valid static storage.
+unsafe fn install_handler(signal: c_int, prev_handler: *mut Option<sigaction>) {
+    // SAFETY: All operations in this block are guarded by unsafe blocks
+    // and are part of the signal handler installation process
+    unsafe {
+        let mut sa: sigaction = std::mem::zeroed();
+        sa.sa_sigaction = crash_signal_handler as usize;
+        sa.sa_flags = SA_SIGINFO | SA_RESTART;
+        libc::sigemptyset(&mut sa.sa_mask);
```

Comment:
I am not sure how this works, but is setting the signal handler before retrieving the previous the way it works?